test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/Package/Download/PackageDownloadRunnerTests.cs

```diff
@@ -603,6 +563,168 @@ public async Task RunAsync_WithSourceMapping(
         }
     }
 
+    [Fact]
+    public async Task RunAsync_WhenPsmMapsToInsecureSource_AndAllowInsecureConnectionsFalse_LogsErrorForMappedSource()
+    {
+        // Arrange
+        using var context = new SimpleTestPathContext();
+        var insecureSourceUrl = "http://contoso.local/v3/index.json";
+
+        context.Settings.AddSource("InsecureMapped", insecureSourceUrl);
+        context.Settings.AddPackageSourceMapping("InsecureMapped", "Contoso.*");
+
+        var settings = Settings.LoadSettingsGivenConfigPaths([context.Settings.ConfigPath]);
+
+        var args = new PackageDownloadArgs
+        {
+            Packages =
+            [
+                new PackageWithNuGetVersion
+                {
+                    Id = "Contoso.Insecure",
+                    NuGetVersion = NuGetVersion.Parse("1.0.0")
+                }
+            ],
+            OutputDirectory = context.WorkingDirectory,
+            AllowInsecureConnections = false,
+        };
+
+        var logger = new Mock<ILoggerWithColor>(MockBehavior.Loose);
+        var packageSources = new List<PackageSource>
+        {
+            new(insecureSourceUrl, "InsecureMapped")
+        };
+
+        // Act
+        var exit = await PackageDownloadRunner.RunAsync(
+            args,
+            logger.Object,
+            packageSources,
+            settings,
+            CancellationToken.None);
+
+        // Assert
+        exit.Should().Be(PackageDownloadRunner.ExitCodeError);
+    }
+
+    [Fact]
+    public async Task RunAsync_WhenPsmMapsToSecureLocalSource_DoesNotLogErrorEvenWhenOtherSourceIsHttp()
+    {
+        // Arrange
+        using var context = new SimpleTestPathContext();
+        var localSourcePath = context.PackageSource;
+        var insecureSourceUrl = "http://contoso.other/v3/index.json";
```

Comment:
other is not a reserved TLD, so in theory it could be a real domain. If the implementation has a bug, we don't want it to make HTTP requests to some 3rd party server. Please only use known invalid domains, like `.test`.